examples/nnp_training.py

```diff
@@ -218,19 +239,27 @@
 ])
 
 ###############################################################################
-# The way ANI trains a neural network potential looks like this:
-#
-# Phase 1: Pretrain the model by minimizing MSE loss
-#
-# Phase 2: Train the model by minimizing the exponential loss, until validation
-# RMSE no longer improves for a certain steps, decay the learning rate and repeat
-# the same process, stop until the learning rate is smaller than a certain number.
+# Setting up a learning rate scheduler to do learning rate decay
+scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, factor=0.5, patience=100)
+
+###############################################################################
+# Train the model by minimizing the mase loss, until validation RMSE no longer
+# improves during a certain number of steps, decay the learning rate and repeat
+# the same process, stop until the learning rate is smaller than a threshold.
 #
-# We first read the checkpoint files to find where we are. We use `latest.pt`
-# to store current training state. If `latest.pt` does not exist, this
-# this means the pretraining has not been finished yet.
+# We first read the checkpoint files to restart training. We use `latest.pt`
+# to store current training state.
 latest_checkpoint = 'latest.pt'
-pretrained = os.path.isfile(latest_checkpoint)
+
+###############################################################################
+# Resume training from previously saved checkpoints:
+if os.path.isfile(latest_checkpoint):
+    checkpoint = torch.load(latest_checkpoint)
+    model.load_state_dict(checkpoint['nn'])
+    optimizer.load_state_dict(checkpoint['optimizer'])
+
+    if 'scheduler' in checkpoint:
```

Comment:
Since there is no more pretraining, we are always saving checkpoints for scheduler, so this if now is always true.